### PR TITLE
use `azsdkEslint.config` helper in eslint.config.mjs

### PR DESCRIPTION
**Pull request by jeremymeng**
This is a follow-up of JS repo PR https://github.com/Azure/azure-sdk-for-js/pull/32073

### CODE REVIEW

File: packages/autorest.typescript/test/rlcIntegration/generated/bodyComplexRest/eslint.config.mjs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import azsdkEslint from "@azure/eslint-plugin-azure-sdk";
 
-export default [
-  ...azsdkEslint.configs.recommended,
+export default azsdkEslint.config([
```
Comment (MaryGao) [inline packages/autorest.typescript/test/rlcIntegration/generated/bodyComplexRest/eslint.config.mjs:3]: Do we need to update the HLC also?

Comment (jeremymeng) [reply packages/autorest.typescript/test/rlcIntegration/generated/bodyComplexRest/eslint.config.mjs:3]: yes for azure js repo

**Review by MaryGao — APPROVED** [review]

File: packages/autorest.typescript/test/rlcIntegration/generated/bodyComplexRest/eslint.config.mjs (hunk 1 above)
Comment (v-jiaodi) [reply packages/autorest.typescript/test/rlcIntegration/generated/bodyComplexRest/eslint.config.mjs:3]: HLC does not include `eslint.config.mjs`

File: packages/typespec-ts/test/integration/generated/special-headers/repeatability/eslint.config.mjs [hunk 2]
```diff
@@ -1,17 +0,0 @@
-import azsdkEslint from "@azure/eslint-plugin-azure-sdk";
```
[the remark this answers is not in the document]
Comment (v-jiaodi) [reply packages/typespec-ts/test/integration/generated/special-headers/repeatability/eslint.config.mjs:1]: Mistakenly brought in in this PR ：https://github.com/Azure/autorest.typescript/pull/2754

**Comment by jeremymeng** [comment]
@qiaozha @MaryGao @v-jiaodi thanks for helping merge this!

**Pull request merged**